### PR TITLE
Adding Guardrails version to public swagger

### PR DESCRIPTION
**Pull request by sallasindhuja**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1]  
[ARM PR review queue for azure-rest-api-specs repo](https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AWaitForARMFeedback+-label%3AIDCDevDiv+draft%3Afalse+)  
  [ARM PR review queue for azure-rest-api-specs-pr repo](https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AWaitForARMFeedback+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc)  
The PRs are processed last (bottom) to first. Your PR may be on 2nd or later page.  
If your PR is not on the queue, remove the `ARMChangesRequested` label. Most of the time automation should do it, but you can do it manually if necessary.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers](https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750) in pinned Teams announcement  
[4]  
[ARM PR merge queue for azure-rest-api-specs repo](https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AIDCDevDiv+draft%3Afalse)  
  [ARM PR merge queue for azure-rest-api-specs-pr repo](https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc)  

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

### CODE REVIEW

**Review by tbarnes94 — APPROVED** [review]

**Comment by FumingZhang** [comment]
Seeking for an ARM review and approval for this change as it's adding a new resource type `locations/guardrailsVersions` (in the upcoming new API version 2023-07-02-preview), which is a prerequisite for the change in ARM manifest. This is part of the AKS 2023-07-02-preview swagger changes, other changes would be reviewed later based on branch [FumingZhang-containerservice-Microsoft.ContainerService-2023-07-02-preview](https://github.com/Azure/azure-rest-api-specs/tree/FumingZhang-containerservice-Microsoft.ContainerService-2023-07-02-preview).

As for the failed CI [Breaking Change(Cross-Version)](https://github.com/Azure/azure-rest-api-specs/pull/25085/checks?check_run_id=15539631529), it's a false alert as it's comparing this change (a new preview API 2023-07-02-preview) with previous stable API (2023-06-01).

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/ListGuardrailsVersions.json
Comment (weinong) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/ListGuardrailsVersions.json:12]: this should be /subscriptions/subid1/providers/Microsoft.ContainerService/locations/location1/guardrailsVersions/v1.0.0?

Comment (matthchr) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/ListGuardrailsVersions.json:12]: but the ID of each should be shown. @weinong is correct here.

Comment (matthchr) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/ListGuardrailsVersions.json:12]: Basically the list/get payloads should be identical except that the list has the `value` collection. Other than that though, there's no difference between a list and a get in terms of what they show. Get shows the full ID (including `v1.0.0`) and so does list

Comment (sallasindhuja) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/ListGuardrailsVersions.json:12]: Ohh I see. Ok, let me make the changes

Comment (sallasindhuja) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/ListGuardrailsVersions.json]: Yeah, makes sense. I was assuming that we were talking about list path. My bad. The response should have version. I made the changes. Pls take a look.

**Review by weinong — APPROVED** [review]
LGTM

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:6988]: To solve the `systemdata` linter warning, you can refer to ` "$ref": "../../../../../../common-types/resource-management/v3/types.json#/definitions/Resource` rather than `SubResource`. The only difference is that the common type has `systemData` too.

We aren't returning that right now in GET, but we probably could in the future.

**Comment by matthchr** [comment]
For the boolean linter warning, I think we are probably not going to change that. A version either is default or is not, it makes some sense for that to be a simple boolean.

**Comment by FumingZhang** [comment]
@sallasindhuja Seems you'll need to get a formal approval for the breaking change in this PR first, before proceeding with the ARM review

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json
Comment (sallasindhuja) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json]: That makes sense. I made the change.

**Comment by sallasindhuja** [comment]
> @sallasindhuja Seems you'll need to get a formal approval for the breaking change in this PR first, before proceeding with the ARM review

I am going a little back and forth on this. AFAIK, breaking change is not applicable here because this is preview. Also, I am trying to merge my branch into FumingZhang-containerservice-Microsoft.ContainerService-2023-07-02-preview branch where the folder is existing, so yes, as per the rule, it results in breaking change (expected behavior but false alert in our specific case). However, this won't be arising while FumingZhang-containerservice-Microsoft.ContainerService-2023-07-02-preview branch merges into main. So, I wonder if I still need to go through breaking change approval process?

cc: @matthchr @NickKeller

**Comment by matthchr** [comment]
@sallasindhuja is correct. There is no breaking change here. Examine her changes and examine the things the breaking change tooling is flagging. None of the things the breaking change tooling flags are in her PR. Her PR is a pure addition of a new route + resource to an un-released API version (2023-07-02-preview (in `FumingZhang-containerservice-Microsoft.ContainerService-2023-07-02-preview` branch).

**Comment by sallasindhuja** [comment]
@rkmanda I was able to get the approval for breaking change. Could you pls review/approve when you get a chance?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json
Comment (ms-zhenhua) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:6998]: >[] [](http://example.com/codeflow?start=30&length=2)

please follow [http://azure.github.io/autorest/extensions/#x-ms-identifiers](http://azure.github.io/autorest/extensions/#x-ms-identifiers) to add the identifying property

Comment (ms-zhenhua) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:6998]: According to http://azure.github.io/autorest/extensions/#x-ms-identifiers, if you use `id` as the identifying property, you do not need to add `x-ms-identifiers`. `[]` means there is no appropriate identifying properties.

**Comment by ms-zhenhua** [comment]
Please resolve the branch conflicts before reviewing.

**Pull request merged**
